src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SPLIT_REGEXP = "\\W+";
+    private static final String WORD_TO_DETECT_SUPPLY_COUNTER_NUMBER = "supply";
+    private static final String WORD_TO_DETECT_BUY_COUNTER_NUMBER = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String textFromFile = readFromFile(fromFileName);
+        String report = reportCreation(textFromFile);
+        writeToFile(report, toFileName);
+    }
+
+    private String readFromFile(String fromFileName) {
+        File file = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
For the future, do not use such naming(stringBuilder, file, file1), how to guess what it stores